tests/test_updater_delegation_graphs.py

```diff
@@ -425,6 +425,90 @@ def test_hash_bins_graph_traversal(
         finally:
             self.teardown_subtest()
 
+    @dataclass
+    class SuccinctRolesTestCase:
+        bit_length: int
+        target_path: str
+        expected_visited_bin: str
+
+    succinct_bins_graph: utils.DataSet = {
+        "case 1": SuccinctRolesTestCase(
+            bit_length=1,
+            target_path="othermissingpath",
+            expected_visited_bin="delegated-0",
+        ),
+        "case 2": SuccinctRolesTestCase(
+            bit_length=1,
+            target_path="missingpath",
+            expected_visited_bin="delegated-1",
+        ),
+        "case 3": SuccinctRolesTestCase(
+            bit_length=2,
+            target_path="foo",
+            expected_visited_bin="delegated-0",
+        ),
+        "case 4": SuccinctRolesTestCase(
+            bit_length=2,
+            target_path="doo",
+            expected_visited_bin="delegated-1",
+        ),
+        "case 5": SuccinctRolesTestCase(
+            bit_length=2,
+            target_path="missingpath",
+            expected_visited_bin="delegated-2",
+        ),
+        "case 6": SuccinctRolesTestCase(
+            bit_length=2,
+            target_path="bar",
+            expected_visited_bin="delegated-3",
+        ),
+        "many bins case": SuccinctRolesTestCase(
+            bit_length=8,
+            target_path="bar",
+            expected_visited_bin="delegated-fc",
+        ),
+    }
```

Comment:
Very cool test! I must admit thought that it took me a while to understand what it's doing. I have some ideas to make this a bit easier. Feel free to adopt/ignore as you please.

- `expected_visited_bin` implied (to me) that it is some node along the tree traversal, but the tree is actually pretty shallow (root + leaves). Maybe we can just call it  `expected_bin`, as it is the bin we expect the target to fall into.
- Using `delegated-` as name prefix makes sense, but if you call it `bin-` it would further reduce complexity (e.g.: `expected_bin="bin-0"`) 
- Some `target_path` values ("missingpath", "othermissingpath") convey an irrelevant meaning, others don't ("foo", "doo", ...). This is confusing. Maybe you can use all placeholder names?
- A few lines of comments would make my suggestions from above less important. E.g. how `bit_length` determines both the total number of bins and the bin a `target_path` falls into, based on its hash prefix...
- Similarly/alternatively, some of this information could be encoded in the test case names, which are currently not very descriptive, e.g. instead of "case 6" it could say "bin count: 4, target bin index: 3"